src/Engine/Army.java

```diff
@@ -0,0 +1,416 @@
+  }
+  public Army(GameScenario scenario, Commander co)
+  {
+    this(scenario);
+    cos = new Commander[] { co };
+    co.army = this;
+  }
+
+  public void initForGame(GameInstance game)
+  {
+    this.registerForEvents(game);
+    for( Commander co : cos )
+      co.initForGame(game);
+  }
+  public void deInitForGame(GameInstance game)
+  {
+    this.unregister(game);
+    for( Commander co : cos )
+      co.deInitForGame(game);
+  }
+
+  /**
+   * Collect income and handle any COModifiers.
+   * @param map
+   */
+  public GameEventQueue initTurn(MapMaster map)
+  {
+    GameEventQueue events = new GameEventQueue();
+    myView.resetFog();
+
+    money += getIncomePerTurn();
+    for( Commander co : cos )
+    {
+      events.addAll(co.initTurn(map));
+    }
+
+    if( null != aiController )
+    {
+      aiController.initTurn(myView);
+    }
+
+    return events;
+  }
+
+  public void endTurn()
+  {
+    if( aiController != null ) aiController.endTurn();
+    myView.resetFog();
+    for( Commander co : cos )
+      co.endTurn();
+  }
+
+  /**
+   * @return whether these armies would like to kill each other
+   */
+  public boolean isEnemy(Army other)
+  {
+    // If the other army doesn't exist, we can't be friends.
+    if( other == null )
+      return true;
+    // If the other army is us, we can't *not* be friends.
+    if( other == this )
+      return false;
+    // If we have no team, we have no friends.
+    if( team < 0 || other.team < 0 )
+      return true;
+    // If we have teams and we're on the same team...
+    // we're on the same team.
+    if( team == other.team )
+      return false;
+    // Otherwise, we hate each other.
+    return true;
+  }
+  public boolean isEnemy(Commander co)
+  {
+    if( co == null )
+      return true;
+    return isEnemy(co.army);
+  }
+
+  public ArrayList<XYCoord> getOwnedProperties()
+  {
+    ArrayList<XYCoord> output = new ArrayList<>();
+      for( Commander co : cos )
+        output.addAll(co.ownedProperties);
+    return output;
+  }
+  public int getIncomePerTurn()
+  {
+    int total = 0;
+      for( Commander co : cos )
+        total += co.getIncomePerTurn();
+    return total;
+  }
+  public ArrayList<Unit> getUnits()
+  {
+    ArrayList<Unit> output = new ArrayList<>();
+      for( Commander co : cos )
+        output.addAll(co.units);
+    return output;
+  }
+  public boolean canBuyOn(MapLocation loc)
+  {
+    // TODO: maybe calculate whether we have enough money to buy something at this industry
+    final Unit resident = loc.getResident();
+    final Commander co = loc.getOwner();
+    if( null == resident && null != co && co.army == this )
+    {
+      if( co.getShoppingList(loc).size() > 0 )
+      {
+        return true;
+      }
+    }
+    return false;
+  }
+  public int getBuyCost(UnitModel um, XYCoord coordinates)
+  {
+    MapLocation loc = myView.getLocation(coordinates);
+    return loc.getOwner().getBuyCost(um, coordinates);
+  }
+
```

Comment:
What will this do if called on a location owned by nobody (or a different army)?

Reply:
Barf.
I'm not sure what it should do other than that, TBH.